RecoLocalCalo/HcalRecAlgos/src/HFStripFilter.cc

```diff
@@ -0,0 +1,361 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/METReco/interface/HcalPhase1FlagLabels.h"
+#include "CondFormats/HcalObjects/interface/HcalChannelQuality.h"
+
+#include "RecoLocalCalo/HcalRecAlgos/interface/HFStripFilter.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+
+#include <cmath>
+
+HFStripFilter::HFStripFilter(const double stripThreshold, const double maxThreshold,
+                             const double timeMax, const double maxStripTime,
+                             const double wedgeCut, const int seedHitIetaMax, 
+                             const int gap, const int lstrips, const int verboseLevel)
+    : stripThreshold_(stripThreshold),
+      maxThreshold_(maxThreshold),
+      timeMax_(timeMax),
+      maxStripTime_(maxStripTime),
+      wedgeCut_(wedgeCut),
+      seedHitIetaMax_(seedHitIetaMax),
+      gap_(gap),
+      lstrips_(lstrips),
+      verboseLevel_(verboseLevel)
+{
+    // For the description of CMSSW message logging, see
+    // https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger
+    if (verboseLevel_ >= 20)
+        edm::LogInfo("HFStripFilter") << "constructor called";
+}
+
+HFStripFilter::~HFStripFilter()
+{
+    if (verboseLevel_ >= 20)
+        edm::LogInfo("HFStripFilter") << "destructor called";
+}
+
+
+void HFStripFilter::runFilter(HFRecHitCollection& rec,
+                      const HcalChannelQuality* myqual) const
+{
+  if (verboseLevel_ >= 20)
+    edm::LogInfo("HFStripFilter") << "runFilter called";
+  
+  std::vector<HFRecHit> d1strip;
+  std::vector<HFRecHit> d2strip;
+  
+  HFRecHit d1max;
+  HFRecHit d2max;
+  
+  d1max.setEnergy(-10);
+  // find d1 and d2 seed hits with max energy and time < timeMax_
+  for (auto& it : rec)
+    {
+      if (it.time() > timeMax_ || it.time() < 0 || it.energy() < stripThreshold_) continue;	
+      // find HF hit with maximum signal in depth = 1
+      if (it.id().depth() == 1) {
+	if (it.energy() > d1max.energy() && std::abs(it.id().ieta()) < seedHitIetaMax_) {
+	  d1max = it;
+	}
+      }
+    }
+
+  if (d1max.energy() > 0) d1strip.push_back(d1max);
+  
+  int signStripIeta = 0;
+  int stripIphiMax = 0;
+  int stripIetaMax = 0;
+  int stripDepthMax = 0;
+  
+  if (d1max.energy() > 0) {
+    signStripIeta = std::signbit(d1max.id().ieta());
```

Comment:
std::signbit(arg) returns a bool, which is "false" if arg is positive, and "true" if arg is negative
While the logic here is correct, perhaps it could be more convenient to define all those local variables as "bool"